test/cbmc/proofs/run-cbmc-proofs.py

```diff
@@ -0,0 +1,278 @@
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+
+import argparse
+import asyncio
+import logging
+import math
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+import textwrap
+
+
+DESCRIPTION = "Configure and run all CBMC proofs in parallel"
+# Keep this hard-wrapped at 70 characters, as it gets printed verbatim
+# in the terminal. 70 characters stops here -----------------------> |
+EPILOG = """
+This tool automates the process of running `make report` in each of
+the CBMC proof directories. The tool calculates the dependency graph
+of all tasks needed to build, run, and report on all the proofs, and
+executes these tasks in parallel.
+
+The tool is roughly equivalent to doing this:
+
+        litani init --project "FreeRTOS coreMQTT";
+
+        find . -name cbmc-batch.yaml | while read -r proof; do
+            pushd $(dirname ${proof});
+            make _report;
+            popd;
+        done
+
+        litani run-build;
+
+except that it is much faster and provides some convenience options.
+The CBMC CI runs this script with no arguments to build and run all
+proofs in parallel.
+
+The --no-standalone argument omits the `litani init` and `litani
+run-build`; use it when you want to add additional proof jobs, not
+just the CBMC ones. In that case, you would run `litani init`
+yourself; then run `run-cbmc-proofs --no-standalone`; add any
+additional jobs that you want to execute with `litani add-job`; and
+finally run `litani run-build`.
+"""
+
+
+def get_args():
+    pars = argparse.ArgumentParser(
+        description=DESCRIPTION, epilog=EPILOG,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    for arg in [{
+            "flags": ["-j", "--parallel-jobs"],
+            "type": int,
+            "metavar": "N",
+            "help": "run at most N proof jobs in parallel",
+    }, {
+            "flags": ["--no-standalone"],
+            "action": "store_true",
+            "help": "only configure proofs: do not initialize nor run",
+    }, {
+            "flags": ["-p", "--proofs"],
+            "nargs": "+",
+            "metavar": "DIR",
+            "help": "only run proof in directory DIR (can pass more than one)",
+    }, {
+            "flags": ["--project-name"],
+            "metavar": "NAME",
+            "default": "FreeRTOS coreMQTT",
+            "help": "Project name for report. Default: %(default)s",
+    }, {
+
+    # PROJECT SPECIFIC
+    # This flag exists because we wish to use an external SAT solver for this
+    # project by default, both locally and in CI. This flag is provided to make
+    # it easy to invoke the native solver if needed.
+            "flags": ["--use-native-solver"],
+            "action": "store_true",
+            "help": "Use CBMC's native SAT solver instead of kissat (proofs will run for longer)",
```

Comment:
This --use-native-solver is an argument to the script.  Does the new CI invoke this script with this flag?  I expected to see this flag in the CBMCFLAGS definition in the proof Makefiles for the slow proofs needing this solver, but I don't see that here.

Reply:
The new CI does not use this flag. Both the CI and the developer are expected to *not* use this flag. It's provided just in case you'd like to run minisat.

What this is doing is running _all_ proofs with Kissat.